### PR TITLE
JAMES-3369 BodyValues handling

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
`[ERROR] error reading /root/.m2/repository/com/github/fge/throwing-lambdas/0.5.0/throwing-lambdas-0.5.0.jar; zip file is empty`

test this please

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailGetMethodContract.scala [hunk 1]
```diff
@@ -2265,6 +2265,81 @@ trait EmailGetMethodContract {
...
+         |    "sessionState": "75128aab4b1b",
+         |    "methodResponses": [
+         |        [
+         |            "Email/get",
+         |            {
+         |                "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |                "state": "000001",
+         |                "list": [
+         |                    {
+         |                        "id": "1",
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailGetMethodContract.scala:2316]: messageId.serialize ?

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala [hunk 2]
```diff
@@ -253,18 +253,44 @@ class Serializer @Inject() (mailboxIdFactory: MailboxId.Factory) {
...
   private implicit val messageIdWrites: Writes[MessageId] = id => JsString(id.serialize())
-  private implicit def emailWritesWithPropertyFilter(properties: Properties)(implicit partsWrites: Writes[EmailBodyPart]): Writes[Email] = Json.writes[Email]
-    .transform(obj => properties.filter(obj))
-  private implicit def bodyPartWritesWithPropertyFilter(properties: Properties): Writes[EmailBodyPart] = Json.writes[EmailBodyPart]
-    .transform(obj => properties.filter(obj))
+  private implicit def bodyValueMapWrites(implicit bodyValueWriter: Writes[EmailBodyValue]): Writes[Map[PartId, EmailBodyValue]] =
+    mapWrites[PartId, EmailBodyValue](_.value.toString(), bodyValueWriter)
+  private def bodyPartWritesWithPropertyFilter(properties: Properties): Writes[EmailBodyPart] =
+    new Writes[EmailBodyPart] {
+      def removeJsNull(obj: JsObject): JsObject =
```
Comment (remk) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala:271]: Which fields could be null ?

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailGet.scala [hunk 3]
```diff
@@ -38,6 +38,7 @@ case class EmailIds(value: List[UnparsedEmailId])
 
 case class EmailGetRequest(accountId: AccountId,
                            ids: Option[EmailIds],
+                           fetchAllBodyValues: Option[Boolean],
```
Comment (remk) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailGet.scala:51]: all those booleans could be strong typed

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala (hunk 2 above)
Comment (chibenwa) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala:271]: Option[]

**Review by Arsnael — APPROVED** [review]

**Review by remk — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**